app/helpers/settings_helper.rb

```diff
@@ -1,31 +1,35 @@
 module SettingsHelper
-  def value(setting)
-    if setting.readonly?
-      return readonly_field(setting, :value,
-        {:title => _("This setting is defined in the configuration file '%{filename}' and is read-only.") % {:filename => setting.class.config_file}, :helper => :show_value})
-    end
+  def setting_presenter(setting)
+    presenter = Foreman.setting_manager.settings[setting.name]
```

Comment:
What is the preferred way to get the setting we need? Foreman.setting_manager.settings[setting.name] or `Setting[setting.name]`? I think the former bypasses the cache...

Reply:
No, it does not bypass the cache. I've moved the cache to the `SettingManager` so no they are equal, I'd probably keep `Setting[setting.name]` as preferred way to get setting value and that is just a proxy for `settin_manager[setting.name]`. But `Foreman.setting_manager.settings[setting.name]` is accessing the setting registry, where we keep all info about the setting and returns the `Presenter`.
It's probably bit confusing and we can drop `SettingManager#[]` and call `SettingManager#read_setting` from `Setting[]` directly.